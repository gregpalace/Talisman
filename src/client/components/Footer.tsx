import React from 'react';

import { VERSION } from '../constants/actionTypes';

import '../../stylesheets/footer.scss';

const Footer = () => {
  return (
    <footer className='footer'>
      <span>{`Talisman ~ v${VERSION}`} </span>

      <a className='footer-link' href='https://github.com/gregpalace/Talisman'>
        {'View on Github'}
      </a>
    </footer>
  );
};

export default Footer;
