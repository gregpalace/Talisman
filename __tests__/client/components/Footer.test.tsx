import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Footer from '../../../src/client/components/Footer';

describe('React unit tests', () => {
  let wrapper: any;

  beforeAll(() => {
    wrapper = shallow(<Footer />);
  });

  describe('<Footer />', () => {
    it('should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a <footer/> with information about the game', () => {
      expect(wrapper.find('footer')).toBeDefined();
      expect(wrapper.find('span').text()).toBe('Talisman ~ v1 ');
      expect(wrapper.find('a').prop('className')).toBe('footer-link');
      expect(wrapper.find('a').text()).toBe('View on Github');
    });
  });
});
