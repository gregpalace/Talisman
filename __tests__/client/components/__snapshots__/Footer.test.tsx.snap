// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <Footer /> should render correctly 1`] = `
<footer
  className="footer"
>
  <span>
    Talisman ~ v1
     
  </span>
  <a
    className="footer-link"
    href="https://github.com/gregpalace/Talisman"
  >
    View on Github
  </a>
</footer>
`;
